be/src/exec/vectorized/jdbc_scanner.cpp

```diff
@@ -0,0 +1,515 @@
+        }                                                                                                       \
+        column->append_nulls(1);                                                                                \
+        return Status::OK();                                                                                    \
+    }
+
+JDBCScanner::~JDBCScanner() {}
+
+Status JDBCScanner::open(RuntimeState* state) {
+    _jni_env = JVMFunctionHelper::getInstance().getEnv();
+    if (_jni_env == nullptr) {
+        return Status::InternalError("Cannot get jni env");
+    }
+
+    RETURN_IF_ERROR(_init_jdbc_bridge());
+
+    RETURN_IF_ERROR(_init_jdbc_scan_context());
+
+    RETURN_IF_ERROR(_init_jdbc_scanner());
+
+    RETURN_IF_ERROR(_init_column_class_name());
+    return Status::OK();
+}
+
+Status JDBCScanner::get_next(RuntimeState* state, ChunkPtr* chunk, bool* eos) {
+    *chunk = std::make_shared<Chunk>();
+    std::vector<SlotDescriptor*> slot_descs = _tuple_desc->slots();
+
+    // init column information
+    for (auto& slot_desc : slot_descs) {
+        ColumnPtr column = ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
+        (*chunk)->append_column(std::move(column), slot_desc->id());
+    }
+    bool has_next = false;
+    RETURN_IF_ERROR(_has_next(&has_next));
+    if (!has_next) {
+        *eos = true;
+        return Status::OK();
+    }
+    int chunk_size = state->chunk_size();
+    jobject jchunk;
+    DeferOp defer([&jchunk, this] () {
+        if (jchunk != nullptr) {
+            _jni_env->DeleteLocalRef(jchunk);
+        }
+    });
+
+    RETURN_IF_ERROR(_get_next_chunk(chunk_size, &jchunk));
+    RETURN_IF_ERROR(_fill_chunk(jchunk, chunk));
+    return Status::OK();
+}
+
+Status JDBCScanner::close(RuntimeState* state) {
+    return _close_jdbc_scanner();
+}
+
+Status JDBCScanner::_init_jdbc_bridge() {
+    // 1. construct JDBCBridge
+    _jdbc_bridge_cls = _jni_env->FindClass("com/starrocks/jdbcbridge/JDBCBridge");
+    DCHECK(_jdbc_bridge_cls != nullptr);
+
+    jmethodID constructor = _jni_env->GetMethodID(_jdbc_bridge_cls, "<init>", "()V");
+    DCHECK(constructor != nullptr);
+
+    _jdbc_bridge = _jni_env->NewObject(_jdbc_bridge_cls, constructor);
+
+    // 2. set class loader
+    jmethodID set_class_loader = _jni_env->GetMethodID(_jdbc_bridge_cls, "setClassLoader", "(Ljava/lang/String;)V");
+    DCHECK(set_class_loader != nullptr);
+
+    jstring driver_location = _jni_env->NewStringUTF(_scan_ctx.driver_path.c_str());
+    _jni_env->CallVoidMethod(_jdbc_bridge, set_class_loader, driver_location);
+    _jni_env->DeleteLocalRef(driver_location);
+    CHECK_JAVA_EXCEPTION("set class loader failed")
+
+    return Status::OK();
+}
+
+Status JDBCScanner::_init_jdbc_scan_context() {
+    jclass scan_context_cls = _jni_env->FindClass("com/starrocks/jdbcbridge/JDBCScanContext");
```

Comment:
All class names should be constant variable.

Reply:
> All class names should be constant variable.

fixed